File: data/transcoder_evaluation_gfg/go/SUM_K_TH_GROUP_ODD_POSITIVE_NUMBERS.go
package main

import (
		"fmt"
	"os"
	"unsafe"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}
func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
func cmpfunc(a unsafe.Pointer, b unsafe.Pointer) int {
	return *(*int)(a) - *(*int)(b)
}


func f_gold(k int) int {
	var (
		cur int = (k * (k - 1)) + 1
		sum int = 0
	)
	for func() int {
		p := &k
		x := *p
		*p--
		return x
	}() != 0 {
		sum += cur
		cur += 2
	}
	return sum
}
//TOFILL
func main() {
	var (
		n_success int     = 0
		param0    []int = []int{91, 52, 78, 51, 65, 39, 42, 12, 56, 98}
	)
	for i := int(0); i < len(param0[:]); i++ {
		if f_filled(param0[i]) == f_gold(param0[i]) {
			n_success += 1
		}
	}
	fmt.Print("#Results:", " ", n_success, ", ", len(param0[:]))
	os.Exit(0)
}

File: data/transcoder_evaluation_gfg/rust/SUM_K_TH_GROUP_ODD_POSITIVE_NUMBERS.rs
#![feature(main)]
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//

#[no_mangle]
pub unsafe fn abs(mut x: i32)
 -> i32 {
    return x.abs();
}

#[no_mangle]
pub unsafe fn min(mut x: i32, mut y: i32)
 -> i32 {
    return if x < y { x } else { y };
}
#[no_mangle]
pub unsafe fn max(mut x: i32, mut y: i32)
 -> i32 {
    return if x > y { x } else { y };
}

pub unsafe fn f_gold(mut k: i32) -> i32 {
    let mut cur: i32 = k * (k - 1 as i32) + 1 as i32;
    let mut sum: i32 = 0 as i32;
    loop  {
        let fresh0 = k;
        k = k - 1;
        if !(fresh0 != 0) { break ; }
        sum += cur;
        cur += 2 as i32
    }
    return sum;
}
//TOFILL
unsafe fn main_0() -> i32 {
    let mut n_success: i32 = 0 as i32;
    let mut param0: [i32; 10] =
        [91 as i32, 52 as i32, 78 as i32,
         51 as i32, 65 as i32, 39 as i32,
         42 as i32, 12 as i32, 56 as i32,
         98 as i32];
    let mut i: i32 = 0 as i32;
    while i < param0.len() as i32 {
        if f_filled(param0[i as usize]) == f_gold(param0[i as usize]) {
            n_success += 1 as i32
        }
        i += 1
    }
    println!("{} {} {} {} {}", "#Results:", " ", n_success, ", ", param0.len() as i32);
    return 0 as i32;
}
#[main]
pub fn main() { unsafe { ::std::process::exit(main_0() as i32) } }
